server/matchmaker/algorithm.py

```diff
@@ -42,39 +31,44 @@ def _unmatch(self, s1: Search):
 
 class StableMarriage(MatchmakingPolicy):
     def find(self) -> Dict[Search, Search]:
-        """Perform SM_NUM_TO_RANK runs of the stable matching algorithm. 
-        Assumes that _MatchingGraph.build_sparse() only returns edges whose matches are acceptable
-        to both parties."""
-        ranks = _MatchingGraph.build_sparse(self.searches)
+        """ Perform the stable matching algorithm until a maximal stable matching
+        is found. Assumes that _MatchingGraph.build_weighted() only returns edges
+        whose matches are acceptable to both parties.
+        """
+        ranks = _MatchingGraph.build_weighted(self.searches)
+        _MatchingGraph.remove_isolated(ranks)
+        max_degree = max((len(edges) for edges in ranks.values()), default=0)
+
         self.matches.clear()
 
-        for i in range(SM_NUM_TO_RANK):
+        for i in range(max_degree):
             self._logger.debug(
                 "Round %i of stable marriage, currently %i matches", i,
                 len(self.matches) // 2
             )
             # Do one round of proposals
-            if len(self.matches) == len(self.searches):
+            if len(self.matches) == len(ranks):
                 # Everyone found a match so we are done
                 break
 
-            for search in self.searches:
+            for search in ranks:
                 if search in self.matches:
                     continue
 
                 if not ranks[search]:
                     # Preference list exhausted
```

Comment:
as above maybe
```python
preference_list_exhausted = len(ranks[search]) == 0
if preference_list_exhausted:
  continue
```

Reply:
If this calculations are not in loop, i don't see the reason for that.

Reply:
Well, the reason would be convey the information from the comment directly in the code. This way you can avoid having a stale comment. Because if someone ever does change the code, then they will remember to rename the variable if needed, but they might not remember to update the comment. 

Personally, I don't really like it though. I think that solution is pretty ugly and I'd prefer the comment. Usually I try to avoid comments, but I think if you're going to put a comment anywhere in your code, it should be in the part with the complicated algorithm. I see the intent, but I prefer the comment in this situation.